EIPS/eip-7007.md

```diff
@@ -0,0 +1,160 @@
+
+```solidity
+pragma solidity ^0.8.18;
+
+/**
+ * @title ERC7007 Token Standard, optional enumeration extension
+ * Note: the ERC-165 identifier for this interface is 0xfa1a557a.
+ */
+interface IERC7007Enumerable is IERC7007 {
+    /**
+     * @dev Returns the token ID given `prompt`.
+     */
+    function tokenId(bytes calldata prompt) external view returns (uint256);
+
+    /**
+     * @dev Returns the prompt given `tokenId`.
+     */
+    function prompt(uint256 tokenId) external view returns (string calldata);
+}
+```
+
+ERC-7007 Metadata JSON Schema for reference
+
+```json
+{
+    "title": "AIGC Metadata",
+    "type": "object",
+    "properties": {
+        "name": {
+            "type": "string",
+            "description": "Identifies the asset to which this NFT represents"
+        },
+        "description": {
+            "type": "string",
+            "description": "Describes the asset to which this NFT represents"
+        },
+        "image": {
+            "type": "string",
+            "description": "A URI pointing to a resource with mime type image/* representing the asset to which this NFT represents. Consider making any images at a width between 320 and 1080 pixels and aspect ratio between 1.91:1 and 4:5 inclusive."
+        },
+
+        "prompt": {
+            "type": "string",
+            "description": "Identifies the prompt from which this AIGC NFT generated"
+        },
+        "aigc_type": {
+            "type": "string",
+            "description": "image/video/audio..."
+        },
+        "aigc_data": {
+            "type": "string",
+            "description": "A URI pointing to a resource with mime type image/* representing the asset to which this AIGC NFT represents."
+        }
+    }
+}
+```
+
+## Rationale
+
+The zkML AIGC-NFTs standard aims to extend the existing [ERC-721](./eip-721.md) token standard to accommodate the unique requirements of AI Generated Content NFTs representing models in a collection. This standard provides interfaces to use zkML to verify whether or not the AIGC data for an NFT is generated from a certain ML model with certain input (prompt). The proposed interfaces allow for additional functionality related to minting, verifying, and enumerating AIGC-NFTs. Additionally, the metadata schema provides a structured format for storing information related to AIGC-NFTs, such as the prompt used to generate the content and the proof of ownership.
+
+With this standard, model owners can publish their trained model and its ZKP verifier to Ethereum. Any user can claim an input (prompt) and publish the inference task, any node that maintains the model and the proving circuit can perform the inference and proving, then submit the output of inference and the ZK proof for the inference trace into the verifier that is deployed by the model owner. The user that initiates the inference task will own the output for the inference of that model and input (prompt).
```

Comment:
Both of these paragraphs seem more like they belong in the motivation section. The rationale section should explain individual technical decisions made within the EIP itself, while the motivation section should justify the EIP as a whole.